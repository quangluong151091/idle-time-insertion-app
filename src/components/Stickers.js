import React from 'react';
import Sticker from './Sticker.js'

class Stickers extends React.Component {
    render() {
        return (
            <td className="side">
                <h3>IDLE STICKERS</h3>
                <table id="sticker_list" className="table-sticker" style={{ width: '100%' }}>
                    <tbody>
                        <tr>
                            <td colSpan="2">
                                <h5>MAINTENANCE</h5>
                            </td>
                        </tr>
                        <tr className="centering">
                            <Sticker id='1' img="enduser" name="End User" />
                            <Sticker id='2' img="primapower" name="Prima Power" />
                        </tr>
                        <tr>
                            <td colSpan="2">
                                <h5>SETUP</h5>
                            </td>
                        </tr>
                        <tr className="centering">
                            <Sticker id='3' img="material" name="Material" />
                            <Sticker id='4' img="tool" name="Tool" />
                        </tr>
                        <tr>
                            <td colSpan="2">
                                <h5>FAILURE</h5>
                            </td>
                        </tr>
                        <tr className="centering">
                            <Sticker id='5' img="ic-icon" name="Operation" />
                            <Sticker id='6' img="ic-icon" name="Tool" />
                        </tr>
                        <tr className="centering">
                            <Sticker id='7' img="ic-icon" name="Material" />
                            <Sticker id='8' img="ic-icon" name="Production order" />
                        </tr>
                        <tr className="centering">
                            <Sticker id='9' img="ic-icon" name="Program" />
                            <td style={{ width: '50%' }}>&nbsp;</td>
                        </tr>
                        <tr>
                            <td colSpan="2">
                                <h5>OTHER</h5>
                            </td>
                        </tr>
                        <tr className="centering">
                            <Sticker id='10' img="ic-icon" name="Other" />
                            <td style={{ width: '50%' }}>&nbsp;</td>
                        </tr>
                    </tbody>
                </table>
            </td>
        );
    }
}

export default Stickers