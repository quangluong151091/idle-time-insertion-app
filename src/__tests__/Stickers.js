import React from 'react';
import { shallow } from 'enzyme';

import Stickers from '../components/Stickers';

describe('Stickers', () => {
  it('should render correctly', () => {

    const component = shallow(<Stickers />);
  
    expect(component).toMatchSnapshot();
  });
});