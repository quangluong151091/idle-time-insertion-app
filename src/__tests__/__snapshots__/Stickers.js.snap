// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stickers should render correctly 1`] = `
<td
  className="side"
>
  <h3>
    IDLE STICKERS
  </h3>
  <table
    className="table-sticker"
    id="sticker_list"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <tbody>
      <tr>
        <td
          colSpan="2"
        >
          <h5>
            MAINTENANCE
          </h5>
        </td>
      </tr>
      <tr
        className="centering"
      >
        <Sticker
          id="1"
          img="enduser"
          name="End User"
        />
        <Sticker
          id="2"
          img="primapower"
          name="Prima Power"
        />
      </tr>
      <tr>
        <td
          colSpan="2"
        >
          <h5>
            SETUP
          </h5>
        </td>
      </tr>
      <tr
        className="centering"
      >
        <Sticker
          id="3"
          img="material"
          name="Material"
        />
        <Sticker
          id="4"
          img="tool"
          name="Tool"
        />
      </tr>
      <tr>
        <td
          colSpan="2"
        >
          <h5>
            FAILURE
          </h5>
        </td>
      </tr>
      <tr
        className="centering"
      >
        <Sticker
          id="5"
          img="ic-icon"
          name="Operation"
        />
        <Sticker
          id="6"
          img="ic-icon"
          name="Tool"
        />
      </tr>
      <tr
        className="centering"
      >
        <Sticker
          id="7"
          img="ic-icon"
          name="Material"
        />
        <Sticker
          id="8"
          img="ic-icon"
          name="Production order"
        />
      </tr>
      <tr
        className="centering"
      >
        <Sticker
          id="9"
          img="ic-icon"
          name="Program"
        />
        <td
          style={
            Object {
              "width": "50%",
            }
          }
        >
           
        </td>
      </tr>
      <tr>
        <td
          colSpan="2"
        >
          <h5>
            OTHER
          </h5>
        </td>
      </tr>
      <tr
        className="centering"
      >
        <Sticker
          id="10"
          img="ic-icon"
          name="Other"
        />
        <td
          style={
            Object {
              "width": "50%",
            }
          }
        >
           
        </td>
      </tr>
    </tbody>
  </table>
</td>
`;
